cloudinit/config/schema.py

```diff
@@ -1170,34 +1203,24 @@ def get_schema_dir() -> str:
     return os.path.join(os.path.dirname(os.path.abspath(__file__)), "schemas")
 
 
-def get_schema() -> dict:
-    """Return jsonschema coalesced from all cc_* cloud-config modules."""
-    # Note versions.schema.json is publicly consumed by schemastore.org.
-    # If we change the location of versions.schema.json in github, we need
-    # to provide an updated PR to
-    # https://github.com/SchemaStore/schemastore.
+def get_schema(schema_type: Optional[str] = "cloud-config") -> dict:
+    """Return jsonschema for a specific type.
 
-    # When bumping schema version due to incompatible changes:
-    # 1. Add a new schema-cloud-config-v#.json
-    # 2. change the USERDATA_SCHEMA_FILE to cloud-init-schema-v#.json
-    # 3. Add the new version definition to versions.schema.cloud-config.json
-    schema_file = os.path.join(get_schema_dir(), USERDATA_SCHEMA_FILE)
+    Return empty schema when no specific schema file exists.
+    """
+    schema_file = os.path.join(
+        get_schema_dir(), SCHEMA_FILES_BY_TYPE[schema_type]["v1"]
```

Comment:
What would happen if we ever had a `v2` version of the schema? Would we have to update this here as well? If so, I'm questioning the value of the "v1" keys on lines 54 and 55. If you think they are valuable, I wonder if it's also worth having a "latest" key.

Reply:
Probably worth defining the `latest` key.  I think we'll need to differentiate v1 from v2 as we continue to support both. But, let's establish a latest alias that'll point to v1 until we add v2. At that point latest == v2.
We will have to change get_schema in a subsquent PR to add an optional preferred version because we do support network v1 and v2. But that work wasn't included in this PR.